### ISSUE TITLE
flow node id

### ISSUE THREAD

**Issue by sychris** [FlydeHQ/flyde#53]
you can right click to set a description but there's no way to set a node: id (which is the default name of the node after its imported) this leads to a few different bugs 1, the default name can't be changed without editing it in text, 2. if 2 flows are made from say hello world example and one is placed in the other you get a fun infinite loop, 3 when you update the inner flow it doesn't reload making it seam like the nodes are frozen.

also manually renaming a node ID after its been placed on another flow breaks that flow resulting in a 

Errors: Flyde editor error: Failed to resolve flow's dependencies: Error: HelloWorld2 in d:\coding_projects\flyde\helloflow.flyde is using referenced node with id old_id that is not imported

**Comment by GabiGrin** [comment]
Indeed, it's on my todo list (which I should migrate to GH issues ASAP) for a long time.

What makes this not a trivial fix is the need to properly refactor dependants. It's not too hard on a single project level, but what if you export your node somewhere else? perhaps that is too far-fetched. 

Also, planning on making the "Failed to resolve flow's dependencies" screen much better so you can solve the conflicts by either pointing to a new location or removing the broken usage.
